### PR TITLE
Feature/6775 devops continous integration sp2

### PR DESCRIPTION
**Pull request by ravi102465**
This PR has changes for CI :
 

1. Assembly variable changes.
2. Code coverage generation and publish

### CODE REVIEW

**Review by WorkRD — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -19,6 +19,14 @@ variables:
...
   - name: BuildCounter
     value: $[counter(format('{0:yyyyMMdd}', pipeline.startTime), 1)]
+  - name: UKHOAssemblyCompany
+    value: "UK Hydrographic Office"
+  - name: UKHOAssemblyVersionPrefix
+    value: "1.0."
+  - name: UKHOAssemblyProduct
+    value: "Exchange Set Service"
+  - name: UKHOAssemblyCopyright
+    value: "Copyright � UK Hydrographic  Office"
```
Comment (WorkRD) [inline azure-pipelines.yml:29]:
```suggestion
    value: "Copyright � UK Hydrographic Office"
```

Comment (ravi102465) [reply azure-pipelines.yml]: Removed extra space

File: Apply-AssemblyVersionAndDefaults.ps1 [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+$UKHOAssemblyCopyright = "$UKHOAssemblyCopyright $(Get-Date -Format "yyyy")"
+
+$assemblyValues = @{
+    "Company"         = $UKHOAssemblyCompany;
+    "Copyright"       = $UKHOAssemblyCopyright;
+    "Description"     = $UKHOAssemblyProduct;
+    "Product"         = $UKHOAssemblyProduct;
+    "AssemblyVersion" = $versionToApply;
+    "FileVersion"     = $versionToApply;
+    "Version"         = $versionToApply;
```
Comment (vijendr13545) [inline Apply-AssemblyVersionAndDefaults.ps1:31]: I was looking into CI build for value of `versionToApply`. 
For example version number generated is `1.0.20210.21` for build https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_build/results?buildId=7537&view=logs&j=4e6201ad-7029-5abf-ddbd-3fcb13fcbeca&t=474823bb-dd3a-55ba-99c4-d7022bf41484 

IMHO version number should be easy to link to it's CI build. I found it difficult to link it back, please can you have a look.

Comment (ravi102465) [reply Apply-AssemblyVersionAndDefaults.ps1:31]: Changes done now version is coming "1.0.20210524.6" where 1.0. is prefix version and 20210524.6 is date and rev for pipeline

Comment (ravi102465) [reply Apply-AssemblyVersionAndDefaults.ps1:31]: Assembly version has limitation that any major, minor, build, revision can store only till 65,535 so yyyymmdd is crossing the limit. I have change build number to $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yy)$(DayOfYear).$(BuildCounter)
now it comes like UKHO.exchange-set-service_merge_21144.8 and same value is added in assembly info

**Review by vijendr13545 — APPROVED** [review]
LGTM

**Review by rockydevnet — APPROVED** [review]
LGTM

**Pull request merged**